src/main/java/seedu/address/logic/parser/SavePdfCommandParser.java

```diff
@@ -0,0 +1,35 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+import seedu.address.logic.commands.SavePdfCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new SavePdfCommand object
+ */
+public class SavePdfCommandParser implements Parser<SavePdfCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the SavePdfCommand
+     * and returns a SavePdfCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SavePdfCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            return new SavePdfCommand(Optional.empty());
+        }
+
+        try {
+            Optional<LocalDate> date = Optional.of(ParserUtil.parseAnyDate(args));
```

Comment:
Any reason with using Optional? Date should always be non null.

Reply:
savepdf has optional date input, so if there's no date input, then ill throw in a optional.empty() as parameter of the constructor